### PR TITLE
fix: rounding to integer currency amounts

### PR DESCRIPTION
**Pull request by UncleSamtoshi**

### CODE REVIEW

**Comment by UncleSamtoshi** [comment]
@sebastienverreault I was a little confused on the wording of the Dealer API. This PR assumes that "Buy" and "Sell" always refer to USD.

**Comment by vindard** [comment]
As I understand it BTC is our reference currency and we're "buying & selling" USD

File: src/app/wallets/send-lightning.ts [hunk 1]
```diff
@@ -607,9 +607,9 @@ const executePaymentViaLn = async ({
         // TODO: test properly. move this to domain
         const baseCentsWithoutFee = toCents(amount)
         const satsWithoutFee = toSats(rawRoute.tokens - rawRoute.safe_fee)
-        const ratio = satsWithoutFee / baseCentsWithoutFee
+        const centsPerSat = baseCentsWithoutFee / satsWithoutFee
```
Comment (dolcalmi) [inline src/app/wallets/send-lightning.ts:610]: if toCents and toSats are only casting then are not needed  in lines 608-609

File: src/services/dealer-price/index.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export const DealerPriceService = (): IDealerPriceService => {
           amountInSatoshis,
         ),
       )
-      return toCents(response.getAmountInCents())
+      return toCents(Math.floor(response.getAmountInCents()))
```
Comment (dolcalmi) [inline src/services/dealer-price/index.ts:98]: if we will persist big int then should be parsed here

File: src/services/dealer-price/index.ts [hunk 3]
```diff
@@ -111,7 +111,7 @@ export const DealerPriceService = (): IDealerPriceService => {
           amountInSatoshis,
         ),
       )
-      return toCents(response.getAmountInCents())
+      return toCents(Math.ceil(response.getAmountInCents()))
```
Comment (dolcalmi) [inline src/services/dealer-price/index.ts:114]: same here

File: src/services/dealer-price/index.ts [hunk 4]
```diff
@@ -131,7 +131,7 @@ export const DealerPriceService = (): IDealerPriceService => {
           .setAmountInSatoshis(amountInSatoshis)
           .setTimeInSeconds(timeToExpiryInSeconds),
       )
-      return toCents(response.getAmountInCents())
+      return toCents(Math.floor(response.getAmountInCents()))
```
Comment (dolcalmi) [inline src/services/dealer-price/index.ts:134]: same here

File: src/services/dealer-price/index.ts [hunk 5]
```diff
@@ -148,7 +148,7 @@ export const DealerPriceService = (): IDealerPriceService => {
           .setAmountInSatoshis(amountInSatoshis)
           .setTimeInSeconds(timeToExpiryInSeconds),
       )
-      return toCents(response.getAmountInCents())
+      return toCents(Math.ceil(response.getAmountInCents()))
```
Comment (dolcalmi) [inline src/services/dealer-price/index.ts:151]: and here

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: src/services/dealer-price/index.ts (hunk 2 above)
Comment (UncleSamtoshi) [reply src/services/dealer-price/index.ts:98]: Are you suggesting returning a UsdPaymentAmount? That change could be made but would expand the scope of the PR.

**Review by dolcalmi — DISMISSED** [review]

**Review by dolcalmi — APPROVED** [review]

**Comment by sebastienverreault** [comment]
> @sebastienverreault I was a little confused on the wording of the Dealer API. This PR assumes that "Buy" and "Sell" always refer to USD.

Yeah cross-currency buy/sell are confusing so I tried to make it explicit in the name of the endpoint what the action was, but if you have suggestions to further clarify I'm all for it. 

I wrote this down to make it clear for myself also: [Price Service](https://github.com/GaloyMoney/dealer/blob/main/PRICE-SERVICE.md).

If the service was for [immediate | future] exchange rates it would be clearer, but that was not the requirements.

**Pull request merged**